crates/l3/starknet-instance/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+use starknet_accounts::SingleOwnerAccount;
+use starknet_ff::FieldElement;
+use starknet_providers::jsonrpc::{HttpTransport, JsonRpcClient};
+use starknet_signers::{LocalWallet, SigningKey};
+use url::Url;
+use common::errors::Error;
+
+pub type LocalWalletSignerMiddleware =
+    SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>;
+
+#[derive(Debug)]
+pub struct StarknetClient {
+    client: LocalWalletSignerMiddleware,
+    url: Url,
+}
+
+impl StarknetClient {
+    pub fn attach(
+        rpc_endpoint: Option<String>,
+        chain_id: Option<String>,
+        priv_key: Option<String>,
+        account_addr: Option<String>,
```

Comment:
why take optional if we're going to throw an error if they're not present

Reply:
oh yeah, that will be better, thank you!